File: repos_estudiantes_TARQUI_tercera_entrega/ILLANES_ESCOBAR_FERNANDO_JOHSUA/GENERECIDAD/Ejercicio4/Java/Vector.java
package GENERECIDAD.Ejercicio4.Java;

import java.util.ArrayList;
import java.util.Collections;

public class Vector<T extends Comparable<T>> {
    private ArrayList<T> elementos;

    public Vector() {
        elementos = new ArrayList<>();
    }

    public void agregar(T valor) {
        elementos.add(valor);
    }

    public T obtener(int i) {
        if (i >= 0 && i < elementos.size()) {
            return elementos.get(i);
        }
        throw new IndexOutOfBoundsException("Índice inválido");
    }

    public T valorMayor() {
        return Collections.max(elementos);
    }

    public T valorMenor() {
        return Collections.min(elementos);
    }
}

File: repos_estudiantes_TARQUI_tercera_entrega/ILLANES_ESCOBAR_FERNANDO_JOHSUA/GENERECIDAD/Ejercicio4/Python/Vector.py
from typing import TypeVar, Generic, List

T = TypeVar('T')

class Vector(Generic[T]):
    def __init__(self):
        self.elementos: List[T] = []

    def agregar(self, valor: T):
        self.elementos.append(valor)

    def obtener(self, i: int) -> T:
        if 0 <= i < len(self.elementos):
            return self.elementos[i]
        raise IndexError("Índice fuera de rango")

    def valor_mayor(self) -> T:
        return max(self.elementos)

    def valor_menor(self) -> T:
        return min(self.elementos)

vector = Vector[int]()
vector.agregar(3)
vector.agregar(9)
vector.agregar(1)

print("Elemento en posición 1:", vector.obtener(1))
print("Mayor:", vector.valor_mayor())
print("Menor:", vector.valor_menor())
